### PR TITLE
Add support for ghc-9.0.1

### PR DESCRIPTION
**Pull request by shayne-fletcher**
This PR adds support for ghc-9.0.1 (currently in alpha).

### CODE REVIEW

File: ghc-lib-gen/src/Ghclibgen.hs [hunk 1]
```diff
@@ -223,22 +226,18 @@ compilerDependencies ghcFlavor =
     , "primop-vector-tys.hs-incl"
     , "primop-vector-uniques.hs-incl"
     ] ++
-    [ x | ghcFlavor == GhcMaster
-        , x <- [ "primop-docs.hs-incl"
-               , "GHC/Platform/Constants.hs"
-               ]
-    ]
+    [ "primop-docs.hs-incl" | ghcFlavor `elem` [ GhcMaster, Ghc901 ] ]
```
Comment (cocreature) [inline ghc-lib-gen/src/Ghclibgen.hs:229]: What happened with `Constants.hs`?

**Review by cocreature — APPROVED** [review]
Nice! One question

File: ghc-lib-gen/src/Ghclibgen.hs (hunk 1 above)
Comment (shayne-fletcher) [reply ghc-lib-gen/src/Ghclibgen.hs:229]: It was being listed twice (and I judged was out of place in) `compilerDependencies`. It remains in `packageCode`.

**Pull request merged**